Add 1D Column Hydrostatic Balance and Ekman Layer examples documentation (#2298)
CliMA/ClimaCore.jl: docs/src/examples.md

## 1D Column examples

> ## 2D Cartesian examples

Reviewer: Hi @Snowdog85123 , thank you for updating the file and posting the screenshots of the rendered documentation. You should have kept everything that was in the original section under the header [`2D Cartesian examples`](https://clima.github.io/ClimaCore.jl/dev/examples/#2D-Cartesian-examples) in the original file though. Now, it seems like you are only including your contributions and you accidentally deleted the existing parts. Thank you

The 1D vertical Ekman layer simulation in [`examples/column/ekman.jl`](https://github.com/CliMA/ClimaCore.jl/blob/main/examples/column/ekman.jl) demonstrates the simulation of atmospheric boundary layer dynamics, specifically the Ekman spiral phenomenon resulting from the balance between Coriolis force and vertical diffusion.

#### Equations and discretizations

#### Momentum

Follows the momentum equations with vertical diffusion and Coriolis force:
```math
\begin{align}
\frac{\partial u}{\partial t} &= \frac{\partial}{\partial z}\left(\nu \frac{\partial u}{\partial z}\right) + f(v - v_g) - w\frac{\partial u}{\partial z} \\
\frac{\partial v}{\partial t} &= \frac{\partial}{\partial z}\left(\nu \frac{\partial v}{\partial z}\right) - f(u - u_g) - w\frac{\partial v}{\partial z}
\end{align}
```

These are discretized using the following:
```math
\begin{align}
\frac{\partial u}{\partial t} &\approx D_{f2c}\left(\nu G_{c2f}(u)\right) + f(v - v_g) - A(w, u) \\
\frac{\partial v}{\partial t} &\approx D_{f2c}\left(\nu G_{c2f}(v)\right) - f(u - u_g) - A(w, v)
\end{align}
```

#### Prognostic variables

> * `u`: _horizontal velocity (east-west component)_ measured in m/s.

Reviewer: Here is a suggestion using terminology that is more familiar to this community (please see this [Wiki](https://en.wikipedia.org/wiki/Zonal_and_meridional_flow) reference)

```suggestion
* `u`: _horizontal velocity (zonal or east-west component)_ measured in m/s.
```

> * `v`: _horizontal velocity (north-south component)_ measured in m/s.

Reviewer: Similar to the above:

```suggestion
* `v`: _horizontal velocity (meridional or north-south component)_ measured in m/s.
```

* `w`: _vertical velocity_ measured in m/s (set to zero in this example).

#### Differentiation operators

Because this is a 1D vertical problem, we utilize the staggered vertical grid with:

- `G_{c2f}` is the [gradient operator from cell centers to faces](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.GradientC2F), called `gradc2f` in the example code.
- `D_{f2c}` is the [divergence operator from faces to centers](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.DivergenceF2C), called `divf2c` in the example code.

> - `A` is the [advection operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.AdvectionC2C), called `A` in the example code (though vertical advection is set to zero in this example).

Reviewer: What do you mean by 
> (though vertical advection is set to zero in this example)

?

#### Problem flow and set-up

> This test case is set up in a vertical column domain `[0, L]` with height `L = 200m`. The Coriolis parameter is set to `f = 5e-5 s⁻¹`, the viscosity coefficient to `ν = 0.01 m²/s`, and the geostrophic wind to `(u_g, v_g) = (1.0, 0.0) m/s`.

Reviewer: Hi @Snowdog85123 , you can actually use math mode here, i.e., `$` instead of code mode. Thank you!

The Ekman depth parameter is calculated as `d = sqrt(2 * ν / f)`, which determines the characteristic depth of the boundary layer.

The initial condition is set to a uniform wind profile equal to the geostrophic wind throughout the vertical column. The simulation is run for 50 hours to allow the boundary layer to develop fully.

Boundary conditions are applied as follows:
- At the top boundary: wind velocity equals the geostrophic wind (`u_g`, `v_g`)
- At the bottom boundary: drag condition proportional to the wind speed, where the surface stress is `Cd * |U| * u` and `Cd * |U| * v`

#### Application of boundary conditions

The application of boundary conditions is implemented through the operators:

1. For the u-momentum equation:
   - Top boundary: `Operators.SetValue(FT(ug))` sets u to the geostrophic value
   - Bottom boundary: `Operators.SetValue(Geometry.WVector(Cd * u_wind * u_1))` applies the drag condition

2. For the v-momentum equation:
   - Top boundary: `Operators.SetValue(FT(vg))` sets v to the geostrophic value
   - Bottom boundary: `Operators.SetValue(Geometry.WVector(Cd * u_wind * v_1))` applies the drag condition

> 3. DSS (Direct Stiffness Summation) is applied implicitly through the operators to ensure numerical stability at element boundaries

Reviewer: I don't think we should mention the DSS operator here since it does not appear explicitly in the example. But just for your own reference sake, I would have described it this way:

```suggestion
3. DSS (Direct Stiffness Summation) is applied implicitly through the operators to ensure continuity of the solution at element boundaries and reduction of unnecessary multiplicity
```

#### Equations and discretizations

##### Mass and Potential Temperature

The system maintains hydrostatic balance while solving the continuity equations for density and potential temperature density:

```math
\begin{equation}
  \frac{\partial \rho}{\partial t} = - \nabla \cdot(\rho \boldsymbol{w})
\label{eq:1d-column-continuity}
\end{equation}
```

```math
\begin{equation}
  \frac{\partial \rho\theta}{\partial t} = - \nabla \cdot(\rho\theta \boldsymbol{w})
\label{eq:1d-column-theta}
\end{equation}
```

These are discretized using the following:

```math
\begin{equation}
  \frac{\partial \rho}{\partial t} \approx - D^c_v[I^f(\rho) \boldsymbol{w}]
\label{eq:1d-column-discrete-continuity}
\end{equation}
```

```math
\begin{equation}
  \frac{\partial \rho\theta}{\partial t} \approx - D^c_v[I^f(\rho\theta) \boldsymbol{w}]
\label{eq:1d-column-discrete-theta}
\end{equation}
```

#### Vertical Momentum

The vertical momentum follows:

```math
\begin{equation}
  \frac{\partial \boldsymbol{w}}{\partial t} = -I^f\left(\frac{\rho\theta}{\rho}\right) \nabla^f_v \Pi(\rho\theta) - \nabla^f_v \Phi(z)
\label{eq:1d-column-momentum}
\end{equation}
```

Where:

> - $\Pi(\rho\theta) = C_p \left(\frac{R_d \rho\theta}{p_0}\right)^{\frac{R_m}{C_v}}$ is the Exner function
> - $\Phi(z) = gz$ is the geopotential

Reviewer: ```suggestion
$\Pi(\rho\theta) = C_p \left(\frac{R_d \rho\theta}{p_0}\right)^{\frac{R_m}{C_v}}$ is the Exner function
$\Phi(z) = gz$ is the geopotential
```
